src/FakeItEasy/Creation/CastleDynamicProxy/CastleDynamicProxyGenerator.cs

```diff
@@ -67,14 +89,11 @@ public ProxyGeneratorResult GenerateProxy(Type typeOfProxy, IEnumerable<Type> ad
                 return new ProxyGeneratorResult(DynamicProxyResources.ProxyIsSealedTypeMessage.FormatInvariant(typeOfProxy));
             }
 
+            Guard.AgainstNull(additionalInterfacesToImplement, "additionalInterfacesToImplement");
             GuardAgainstConstructorArgumentsForInterfaceType(typeOfProxy, argumentsForConstructor);
+            Guard.AgainstNull(fakeCallProcessorProvider, "fakeCallProcessorProvider");

```

Comment:
It's the same order as the parameters, but I can change it. Anyway, seeing it again, I find it a bit strange too... I'll make another pass at it, and try to move the guard clauses to the method where the parameters are dereferenced.